doctr/models/predictor/tensorflow.py

```diff
@@ -65,6 +67,13 @@ def __call__(
         origin_page_shapes = [page.shape[:2] for page in pages]
 
         # Detect document rotation and rotate pages
+        if self.detect_orientation:
+            origin_page_orientations = [estimate_orientation(page) for page in pages]
```

Comment:
If detect_orientation=True and straighten_pages=True we should reuse origin_page_orientations otherwise we do it twice

Reply:
Done and pushed. Added also the doc about the detect_orientation param :+1: